neqo-transport/src/recovery/mod.rs

```diff
@@ -654,6 +665,41 @@ impl Loss {
             &mut self.stats.borrow_mut(),
         );
 
+        // When we receive ACKs in Initial space, prime the Handshake space PTO baseline
+        // if we haven't received any Handshake packets yet. This handles the case where:
+        // - Handshake keys are installed (from Server Hello in Initial packets)
+        // - Server's Handshake flight is lost/corrupted
+        // - We haven't sent any Handshake packets yet (nothing to send)
+        // - We haven't received any Handshake packets (largest_acked is None)
+        //
+        // Without this, pto_base_time() returns None for Handshake space and the PTO
+        // timer never arms, causing a deadlock/timeout.
+        //
+        // We need both conditions to avoid false positives:
+        // 1. largest_acked.is_none() - no Handshake packets received yet
+        // 2. Multiple PTOs fired (count > 1) - clearly stuck, not just first retry
+        //
+        // Without count > 1, we'd prime on first Initial ACK even when Handshake
+        // packet is about to arrive, affecting PTO timing in normal flow.
+        if pn_space == PacketNumberSpace::Initial {
```

Comment:
This should be factored out if it's going to be this much.  It's only a few lines, but it's so, so much.

Also, do we need to prime the PTO on ApplicationData when Handshake ACKs are received?